src/write.c

```diff
@@ -2824,8 +2845,8 @@ static avifResult avifEncoderWriteMiniBox(avifEncoder * encoder, avifRWStream *
         AVIF_CHECKRES(avifRWStreamWrite(s, image->xmp.data, image->xmp.size)); // unsigned int(8) xmp_data[xmp_data_size_minus1 + 1];
     }
 
-    const size_t expectedChunkBytes = (hasAlpha ? codecConfigSize : 0) + (hasGainmap ? codecConfigSize : 0) + codecConfigSize +
-                                      image->icc.size + (hasGainmap ? tmapIccSize : 0) + (hasGainmap ? gainmapMetadataSize : 0) +
+    const size_t expectedChunkBytes = codecConfigSize + alphaCodecConfigSize + gainmapCodecConfigSize + image->icc.size +
```

Comment:
Optional/future: This long summation probably should check for integer (`size_t`) overflow. Alternatively we can check for the expected `avifRWStreamOffset(s)` after each `avifRWStreamWrite()` call.

Reply:
This check is already dubious in my opinion. I keep it as a cheap verification but any added complexity would make it not worth it.

> This long summation probably should check for integer (`size_t`) overflow.

It should not be needed as `mini` total size (and thus the sum of its field sizes) is guaranteed to fit in uint32_t per design.

> Alternatively we can check for the expected `avifRWStreamOffset(s)` after each `avifRWStreamWrite()` call.

This would only verify that reading something advance its offset by that many bytes, not that we did not forget something, right?

Reply:
That's correct. I didn't know this check is a double check. I thought it was just checking that the final offset was expected and could be replaced by incremental checks.